flink-connectors/flink-connector-base/src/main/java/org/apache/flink/connector/base/source/hybrid/SwitchedSources.java

```diff
@@ -45,4 +47,8 @@ public SimpleVersionedSerializer<SourceSplit> serializerOf(int sourceIndex) {
     public void put(int sourceIndex, Source source) {
         sources.put(sourceIndex, Preconditions.checkNotNull(source));
     }
+
+    public Integer getNextSubtaskSourceIndex(int previousSubtaskSourceIndex) {
```

Comment:
```suggestion
    public Integer getNextSourceIndex(int prevSourceIndex) {
```

Reply:
This would be misleading as the very purpose is to determine the source index for a specific subtask. Probably it is better to just have `getFirstSourceIndex()` here and keep the rest of the logic in `handleSourceEvent`

Reply:
I agree to change to getFirstSourceInde() and keep the rest of the logic in handleSourceEvent.